pkg/asset/installconfig/aws/ec2.go

```diff
@@ -0,0 +1,36 @@
+package aws
+
+import (
+	"context"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// GetVpcsFromSecurityGroups will get the list of VPCs that are attached to the security groups with the ids provided.
+func GetVpcsFromSecurityGroups(ctx context.Context, session *session.Session, securityGroupIDs []string, region string) ([]string, error) {
+	client := ec2.New(session, aws.NewConfig().WithRegion(region))
+
+	sgIDPtrs := []*string{}
+	for _, sgid := range securityGroupIDs {
+		sgidAlias := sgid
+		sgIDPtrs = append(sgIDPtrs, &sgidAlias)
+	}
+
+	cctx, cancel := context.WithTimeout(ctx, 1*time.Minute)
+	defer cancel()
+
+	sgOutput, err := client.DescribeSecurityGroupsWithContext(cctx, &ec2.DescribeSecurityGroupsInput{GroupIds: sgIDPtrs})
+	if err != nil {
+		return nil, err
+	}
+
+	vpcIds := sets.NewString()
```

Comment:
String sets is deprecated [1]. We should be using the Generics syntax now:
```suggestion
	vpcIds := sets.New[string]()
```

[1] https://pkg.go.dev/k8s.io/apimachinery/pkg/util/sets#String